src/main/java/net/sf/mzmine/modules/peaklistmethods/io/spectraldbsubmit/sorting/MassListSorter.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2006-2018 The MZmine 2 Development Team
+ * 
+ * This file is part of MZmine 2.
+ * 
+ * MZmine 2 is free software; you can redistribute it and/or modify it under the terms of the GNU
+ * General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * MZmine 2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License along with MZmine 2; if not,
+ * write to the Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301
+ * USA
+ */
+
+package net.sf.mzmine.modules.peaklistmethods.io.spectraldbsubmit.sorting;
+
+import java.util.Comparator;
+import net.sf.mzmine.datamodel.DataPoint;
+import net.sf.mzmine.util.ScanUtils;
+
+public class MassListSorter implements Comparator<DataPoint[]> {
+  private double noiseLevel;
+  private ScanSortMode sort;
+
+  public MassListSorter(ScanSortMode sort, double noiseLevel) {

```

Comment:
Why is the noise level necessary here? Mass lists are already filtered. Adding another noise level parameter just creates confusion for users.

Reply:
The noise level is a convenience method. The user does not have to rerun massdetection. 
Especially helpful in the library submit module. The user should take a look at the spectrum and define a resonable noise level to assure the quality of the spectral library entries.